### ISSUE TITLE
Invalid date is not retained in repeated group component

### ISSUE THREAD

**Issue by PallaviGanorkar** [google/android-fhir#1868]
**Describe the bug**
In repeated group component, is user enters invalid date, and then again adds new group, app clears invalid date

**Component**
 SDC library, reference app

**To Reproduce**
1. Go to 'Repeated group Component'
2. Click on 'Add item'
3. "Sample date and dropdown question"  group item is added on screen
4. Enter invalid date and select any option from dropdown
5. Now again Click on 'Add item'
6. See Error "dropdown selected option still persist and invalid date is cleared in first group item"

**Expected behavior**
1. behavior should be consistent for both date and dropdown question and both value should persist when new group item is added

**Screenshots**
[Repeated_Group_issue.webm](https://user-images.githubusercontent.com/13957417/218696023-0e1b08d7-f1a6-4132-a0be-990ff81e60e6.webm)

**Smartphone (please complete the following information):**
 - Device:  Nexus 5X emulator
 - Android version: API level 24

**Comment by santosh-pingle** [comment]
It seems that when the "Add Item" button is clicked, a new reference/object is created for the "questionnaireResponseItemComponent," which does not align with the previous references of "questionnaireResponseItemComponent" that were used to store partial answers against a key in the map.

**Comment by jingtang10** [comment]
@PallaviGanorkar can you please just double check that this is not the case for complete date answers.

**Comment by PallaviGanorkar** [comment]
Yes, I have rechecked it, this is does not observed when valid date is entered

**Comment by jingtang10** [comment]
@santosh-pingle can you doublecheck if this is still an issue

**Comment by santosh-pingle** [comment]
I will check whether issue is reproducible or not.
